### ISSUE TITLE
2.6.7 beta 1 - on automation screen there's mentioned that objectives arent completed.

### ISSUE THREAD

**Issue by RalfCGM** [MilosKozak/AndroidAPS#2736]
I've completed objectives long time ago. Automation screen states that "objectives: objective 11 hasn't started".

I've ran though the objectives while there were still 9 or 10 instead of 11. Maybe that's giving the message?

I've been using automation for a long time now. There doesn't seem to be a check whether there already are automation tasks installed. Maybe, if there are already task installed, autocomplete objective 11?

If I try to pass obj 11, It looks like I have to delete all my automation rules to later add them again.

**Comment by PoweRGbg** [comment]
Objective 11 is just time based. You don't need to remove all your tasks/recipes. Just wait 28 days. :)

**Comment by RalfCGM** [comment]
There's no timer running and in the meantime automation isn't working. Is there any logic in that?

Automation was already working for more then a year,since I completed all objectives before.

I think I have been incorrectly thrown back in the objectives, while not being able to start obj 11 again.

**Comment by guydavies** [comment]
I seem to remember someone saying that Objective 3 had become incomplete
even though all the questions were answered. You might want to check that.
I think when they recommended that (just clicking a single button) it
worked straight away.

**Comment by RalfCGM** [comment]
Indeed my new obj 3 was introduced while I already had completed obj. I'll try.

**Comment by MilosKozak** [comment]
prove to me you have all previous objectives marked as completed. paste screenshots

**Comment by MilosKozak** [comment]
and you need only start Objective 11, not finish

**Comment by RalfCGM** [comment]
I'm finishing obj 3 right now. Hope automation works again afterwards.

 I would propose a warming message after install of the app, that automation was turned off because of unfinished earlier objectives. Some people might rely on automation features which without warning get disabled.

**Comment by RalfCGM** [comment]
Working again after completing all remaining objectives. 

Indeed would propose to add a warming message that automation has been disabled because of not-completed objectives.

**Comment by xJoe** [comment]
In the past we said that for users who have at least completed objective “Starting on an open loop” exam is optional (https://androidaps.readthedocs.io/en/latest/EN/Installing-AndroidAPS/Releasenotes.html#id13)
Does that change now or is it just a "beta feature"?

**Comment by AndyS117** [comment]
+1 on this one. All things running fine but the 2 Automations that I have were imported but getting warning that Objective 11  not started. Turned objectives screen back on and there seems to be no way to start it.

I've been running since before v2 so just pulling my config over each time so perhaps it's just missing there?

**Comment by dansuk** [comment]
+1. Automation disabled here too and like others having been using for some time. Same situation as AndyS117

**Comment by guydavies** [comment]
Yep, I noticed (and reported) this in dev but I have not completed the other Objectives because I am using a virtual pump so cannot start Objective 6.  Been using automation for ages so it’s a bit irritating to lose it.

**Comment by overfrenk** [comment]
Same problem for me too! Goal 10 ended in October 2019 and I was already using automations ... but now they no longer work because goal 11 is not completed! And I can't even eliminate the present automations, but only to create new ones.

**Comment by RalfCGM** [comment]
I think there's an underlying goal for the devs for everyone to complete every objective. After completing obj 2 and 3, you'll be able to start obj 11 and use all your previous automation tiles again. I only adviced to give people a message during first ibstall drat automation was disabled.

**Comment by guydavies** [comment]
That is a very explicit change if true. I've been using AAPS as a decision
support tool with a virtual pump for about a year now and it's had a
significant impact. Automation made switching between weekday and weekend
profiles reliable (I often forget to switch manually). I would say such
rigidity would be a mistake.

**Comment by RalfCGM** [comment]
Completely understand your situation @guydavies ...

**Comment by MilosKozak** [comment]
> In the past we said that for users who have at least completed objective “Starting on an open loop” exam is optional (https://androidaps.readthedocs.io/en/latest/EN/Installing-AndroidAPS/Releasenotes.html#id13)
> Does that change now or is it just a "beta feature"?

from 2.7 all previous objetives must be completed to be able start newer. This rule was not in place before to give everybody time finish it. Open loop is not affected by this change.
@xJoe I alredy updated Release notes. You can review it

**Comment by guydavies** [comment]
Hi @MilosKozak technically, open loopers who were using previous versions *are* affected by this change. I cannot automate changes to my profiles/targets at all now. Those things may not seem important but they increase the value of AAPS for open loop. Without them, the quality of the recommended settings is reduced. 

Please reconsider this final block on automation which had no objectives in any previous versions.

**Comment by xJoe** [comment]
> from 2.7 all previous objetives must be completed to be able start newer. This rule was not in place before to give everybody time finish it. Open loop is not affected by this change.

I assume many long-term users did not understand that they have been given time to finish exam but just did not bother about it.
Shall we inform them (post on FB) now that they will have to finish exam to finish objective 11 in new version @MilosKozak? So they could use time until 2.7 master to complete exam.

**Comment by AndyS117** [comment]
Covering in the release notes, docs and making an announcement in FB group makes sense. This will really only affect people who have been using the system so long that they never completed the quiz part of the objectives because it didn't exist back then.
In my case I know I never saw the exam because I had the objectives turned off since I completed them 2 years ago so figured why did I need it cluttering my screen
